src/TypeSpec.Extension/Emitter.Csharp/src/type/converter.ts

```diff
@@ -36,22 +229,381 @@ export function fromSdkEnumType(
             IsNullable: enumType.nullable,
             Usage: fromUsageFlags(enumType.usage)
         };
-        setUsage(context, enumType.__raw! as Enum, newInputEnumType);
         if (addToCollection) enums.set(enumName, newInputEnumType);
         inputEnumType = newInputEnumType;
     }
     inputEnumType.IsNullable = enumType.nullable; // TO-DO: https://github.com/Azure/autorest.csharp/issues/4314
     return inputEnumType;
 }
 
-export function fromSdkEnumValueType(
+function fromSdkDatetimeType(
+    dateTimeType: SdkDatetimeType
+): InputPrimitiveType {
+    function fromDateTimeKnownEncoding(
+        encoding: DateTimeKnownEncoding
+    ): InputPrimitiveTypeKind {
+        switch (encoding) {
+            case "rfc3339":
+                return InputPrimitiveTypeKind.DateTimeRFC3339;
+            case "rfc7231":
+                return InputPrimitiveTypeKind.DateTimeRFC7231;
+            case "unixTimestamp":
+                return InputPrimitiveTypeKind.DateTimeUnix;
+        }
+    }
+    return {
+        Kind: InputTypeKind.Primitive,
+        Name: fromDateTimeKnownEncoding(dateTimeType.encode),
+        IsNullable: dateTimeType.nullable
+    } as InputPrimitiveType;
+}
+
+function fromSdkDurationType(
+    durationType: SdkDurationType
+): InputPrimitiveType {
+    function fromDurationKnownEncoding(
+        encode: DurationKnownEncoding,
+        wireType: SdkBuiltInType
+    ): InputPrimitiveTypeKind {
+        switch (encode) {
+            case "ISO8601":
+                return InputPrimitiveTypeKind.DurationISO8601;
+            case "seconds":
+                if (wireType.kind === "float" || wireType.kind === "float32") {
+                    return InputPrimitiveTypeKind.DurationSecondsFloat;
+                }
+                return InputPrimitiveTypeKind.DurationSeconds;
+            default:
+                logger.warn(
+                    `invalid encode '${encode}' and wireType '${wireType.kind}' for duration.`
+                );
+                return InputPrimitiveTypeKind.DurationISO8601;
+        }
+    }
+    return {
+        Kind: InputTypeKind.Primitive,
+        Name: fromDurationKnownEncoding(
+            durationType.encode,
+            durationType.wireType
+        ),
+        IsNullable: durationType.nullable
+    } as InputPrimitiveType;
+}
+
+function fromTupleType(tupleType: SdkTupleType): InputIntrinsicType {
+    return {
+        Kind: InputTypeKind.Intrinsic,
+        Name: InputIntrinsicTypeKind.Unknown,
+        IsNullable: tupleType.nullable
+    } as InputIntrinsicType;
+}
```

Comment:
so we map tuple types into `unknown` here
Are we doing the same previously?

Reply:
yes, see: https://github.com/Azure/autorest.csharp/blob/e868777f1a1acbc654ff2db44fc616f2f1613bb0/src/TypeSpec.Extension/Emitter.Csharp/src/lib/model.ts#L364

Reply:
Could you please add a TODO there to indicate that tuples are not officially supported as a note which we could search in the future?